src/calllogger/plugins/internal/beronet.py

```diff
@@ -0,0 +1,93 @@
+__all__ = ["BeroNet"]
+
+# Standard library
+from typing import NoReturn
+import requests
+from requests.auth import HTTPBasicAuth
+import csv
+
+# Local
+from calllogger.plugins import BasePlugin
+from calllogger.record import CallDataRecord as Record
+
+
+class BeroNet(BasePlugin):
+    """
+    Collect call records continuously.
+    Adds support for BeroNet virtual ISDN and VoIP CAPI
+    Collects CDR as a CSV via HTTP API
+    """
+    beronet_ip: str
+    beronet_user: str
+    beronet_password: str
+    beronet_sleep: int = 5
+
+    def collect(self) -> list:
+        result = []
+        auth = HTTPBasicAuth(
+            self.beronet_user,
+            self.beronet_password
+        )
+        command = "TelephonyGetCdr"
+        action = "download"
+        query = f"apiCommand={command}&Action={action}"
+        url = f"http://{self.beronet_ip}/app/api/api.php?{query}"
+
+        response = requests.get(
+            url,
+            auth,
+        )
+        if response.status_code == 200:
+            decoded = response.content.decode('utf-8')
+            cdr = csv.reader(decoded.splitlines(), delimiter=',')
+            result = list(cdr)
+        return result
+
+    def entrypoint(self) -> NoReturn:
+
+        while self.is_running:
+            cdr = self.collect()
+            for call in cdr:
+                if len(call):
+                    direction = call[2]
+                    # Deside if call is Outgoing or Received
+                    if direction.startswith("ISDN"):
+                        self.outgoing(call)
+                    else:
+                        self.received(call)
+            self.timeout.sleep(self.beronet_sleep)
+
+    def outgoing(self, call: list):
+        record = Record(call_type=Record.OUTGOING)
+        record.number = call[5]
+        record.ext = call[4]
+
+        start_date = record.date_str(call[8], fmt="%y/%m/%d-%H:%M:%S")
```

Comment:
record.date_str() dose not work like that. It just sets the parses and sets the record.date for you.

Reply:
You can use "datetime.strptime(date, fmt)" to create a datetime object.

Reply:
Thanks, addressed